### PR TITLE
fix: release resources on break

### LINKED ISSUE #1052: `break` keyword not handled by memory management system
We use `break` for important optimizations in Core (where it's correct, by inspection). 

Ideally we would be able to rule out misuses like the following:

```clojure
(defn f []
  (for [n 0 5]
    (let-do [leak @"LEAK"]
      (break))))
```

### PR DESCRIPTION
**Pull request by hellerve**
This PR fixes #1052 by fixing up `break` statements with the context of which resources need to be released.

This is achieved by descending from any `let` and `while` statement into an enclosed `break` and adding any variables that need to be freed in that scope to its deleters.

Cheers

### CODE REVIEW

File: src/Concretize.hs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE BangPatterns #-}
```
Comment (scolsen) [inline src/Concretize.hs:1]: Seems like no BangPatterns are used?

**Review by scolsen — APPROVED** [review]

File: src/Concretize.hs (hunk 1 above)
Comment (hellerve) [reply src/Concretize.hs]: Good catch! I used this for debugging, but then didn’t need it in the final product.

File: src/Concretize.hs [hunk 2]
```diff
@@ -1262,6 +1268,15 @@ manageMemory typeEnv globalEnv root =
                   Right (XObj (Lst (okF : okArgs)) i t)
         [] -> pure (Right xobj)
     visitList _ = error "Must visit list."
+    searchBreak :: Set.Set Deleter -> XObj -> XObj
```
Comment (eriksvedang) [inline src/Concretize.hs:1271]: I'm confused by this function's name, is there perhaps an alternative that fits it better? Or maybe I just need to be enlightened.

Comment (hellerve) [reply src/Concretize.hs:1271]: `searchForInnerBreak` might be better?

Comment (hellerve) [reply src/Concretize.hs:1271]: Or `fixupInnerBreak`?

Comment (eriksvedang) [reply src/Concretize.hs:1271]: searchForInnerBreak is more clear in my mind

Comment (hellerve) [reply src/Concretize.hs]: Renamed!

**Pull request merged**